cookbook/serverless-voice-chatbot.mdx

```diff
@@ -0,0 +1,189 @@
+
+    def finalize(self):
+        # Finalize resources if needed
+        pass
+```
+### Setting up the Environment
+__Dependencies:__
+- __Objective:__ Ensure all necessary libraries are installed.
+- __Action:__ Run the command below to install dependencies:
+```bash
+pip install torchaudio==2.2.1 soundfile==0.12.1 git+https://github.com/suno-ai/bark.git@refs/pull/391/head faster-whisper==1.0.0 torch==2.2.1 nltk==3.8.1
+```
+This command ensures your environment has all the tools required for the application.
+### Deploying Your Model with Inferless CLI
+1. __Run the following command to initialize your model:__
+```bash
+inferless init
+```
+2. __Upload Custom Runtime:__ Use the following command to upload your [custom runtime](https://github.com/inferless/Voice-Conversational-Chatbot/blob/main/inferless-runtime-config.yaml).
+```bash
+inferless runtime upload
+```
+Here's the custom runtime for the application:
+```yaml
+build:
+  system_packages:
+    - "libssl-dev"
+  python_packages:
+    - "torchaudio==2.2.1"
+    - "soundfile==0.12.1"
+    - "git+https://github.com/suno-ai/bark.git@refs/pull/391/head"
+    - "faster-whisper==1.0.0"
+    - "torch==2.2.1"
+    - "nltk==3.8.1"
+```
+3. __Deploy Model:__ Execute `inferless deploy` to deploy and monitor the build logs on Inferless.
+```bash
+inferless deploy
+```
+### Demo of the Voice Conversational Chatbot.
+<video width="640" height="360" controls>
+  <source src="/videos/voice-chatbot-demo.mp4" type="video/mp4"/>
+  Your browser does not support the video tag.
+</video>
+
+
+### Alternative Deployment Method
+Inferless also supports a user-friendly UI for model deployment, catering to users at all skill levels. Refer to Inferless's documentation for guidance on UI-based deployment.
+## Choosing Inferless for Deployment
+Deploying your Voice Conversational Chatbot application with Inferless offers compelling advantages, making your development journey smoother and more cost-effective. Here's why Inferless is the go-to choice:
+1. __Ease of Use:__ Forget the complexities of infrastructure management. With Inferless, you simply bring your model, and within minutes, you have a working endpoint. Deployment is hassle-free, without the need for in-depth knowledge of scaling or infrastructure maintenance.
+2. __Cold-start Times:__ Inferless's unique load balancing ensures faster cold-starts. Expect around 28.60 seconds to process each queries, significantly faster than many traditional platforms.
+3. __Cost Efficiency:__ Inferless optimizes resource utilization, translating to lower operational costs. Here's a simplified cost comparison:
+
+### Scenario 1
+You are looking to deploy a Voice Conversational Chatbot application for processing 100 queries.<br />
+
+__Parameters:__
+- __Total number of queries:__ 100 daily.<br />
+- __Inference Time:__ All models are hypothetically deployed on A100 80GB, taking 28.60 seconds of processing time and a cold start overhead of 20.72 seconds.<br />
```

Comment:
can we deploy this on A10G

Reply:
@nickaggarwal I have to check that, I have tested it on A100 machine only.